google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -361,37 +338,16 @@ public void onSuccess(PublishResponse result) {
 
           @Override
           public void onFailure(Throwable t) {
-            long nextBackoffDelay =
-                computeNextBackoffDelayMs(outstandingBatch, retrySettings, longRandom);
-
-            if (!isRetryable(t)
-                || retrySettings.getMaxAttempts() > 0
-                    && outstandingBatch.getAttempt() > retrySettings.getMaxAttempts()
-                || System.currentTimeMillis() + nextBackoffDelay
-                    > outstandingBatch.creationTime + retrySettings.getTotalTimeout().toMillis()) {
-              try {
-                ApiException gaxException =
-                    ApiExceptionFactory.createException(
-                        t, GrpcStatusCode.of(Status.fromThrowable(t).getCode()), false);
-                for (OutstandingPublish outstandingPublish :
-                    outstandingBatch.outstandingPublishes) {
-                  outstandingPublish.publishResult.setException(gaxException);
-                }
-              } finally {
-                messagesWaiter.incrementPendingMessages(-outstandingBatch.size());
+            try {
+              ApiException gaxException =
+                  ApiExceptionFactory.createException(
+                      t, GrpcStatusCode.of(Status.fromThrowable(t).getCode()), false);
```

Comment:
Aren't there non-ApiException exceptions that can be thrown?

Reply:
I'm not quite sure what you're getting at. If the `Throwable` isn't `ApiException`, won't `createException` wrap it?

Reply:
The more problematic spot is `Status.fromThrowable(t)` - I doubt that will be well-behaved.

Reply:
Actually, the `Throwable` itself should already be an `ApiException`, right?